api/spec/models/service_spec.rb

```diff
@@ -9,6 +9,38 @@
   it { is_expected.to validate_presence_of :service_specification }
   it { is_expected.to validate_presence_of :service_type }
 
+  describe 'delegated methods' do
+    subject { create :service }
+
+    it { is_expected.to delegate_method(:used_sick_days).to(:used_days_calculator) }
+    it { is_expected.to delegate_method(:used_paid_vacation_days).to(:used_days_calculator) }
+    it { is_expected.to delegate_method(:remaining_sick_days).to(:remaining_days_calculator) }
+    it { is_expected.to delegate_method(:remaining_paid_vacation_days).to(:remaining_days_calculator) }
+  end
+
+  describe 'memoization' do
+    let(:service) { create :service }
+    let(:used_days_calculator) { instance_double ExpenseSheetCalculators::UsedDaysCalculator }
+    let(:remaining_days_calculator) { instance_double ExpenseSheetCalculators::RemainingDaysCalculator }
+
+    before do
+      allow(ExpenseSheetCalculators::UsedDaysCalculator).to receive(:new).and_return used_days_calculator
+      allow(ExpenseSheetCalculators::RemainingDaysCalculator).to receive(:new).and_return remaining_days_calculator
+    end
+
+    it 'creates a used_days_calculator' do
+      service.send(:used_days_calculator)
```

Comment:
Avoid using send

Reply:
Resolved by calling the delegated methods